packages/flutter/lib/src/painting/matrix_utils.dart

```diff
@@ -169,6 +169,60 @@ class MatrixUtils {
     transform = new Matrix4.copy(transform)..invert();
     return transformRect(transform, rect);
   }
+
+  /// Create a transformation matrix which mimics the effects of tangentially
+  /// wrapping the plane on which this transform is applied around a cylinder
+  /// and then looking at the cylinder from a point outside the cylinder.
+  ///
+  /// [radius] simulates the radius of the cylinder the plane is being wrapped
+  /// onto. If the transformation is applied to a 0-dimensional dot instead of
+  /// a plane, the dot would simply translate by +/- [radius] pixels along the
+  /// [orientation] [Axis] when rotating from 0 to +/- 90 degrees.
+  ///
+  /// A positive radius means the object is closest at 0 [angle] and a negative
+  /// radius means the object is closest at [pi] [angle].
+  ///
+  /// [angle] is the difference in angle in radians between the object and the
+  /// viewing point. A positive [angle] on a positive [radius] moves the object
+  /// up when [orientation] is vertical and right when horizontal.
+  ///
+  /// The transformation is always done such that a 0 [angle] keeps the
+  /// transformed object at exactly the same size as before regardless of
+  /// [radius] and [perspective] when [radius] is positive.
+  ///
+  /// [perspective] is a number between 0 and 1 where 0 means looking at the
+  /// object from infinitely far with an infinitely narrow field of view and
+  /// 1 means looking at the object from infinitely close with an infinitely wide
+  /// field of view. Defaults to a sane but arbitrary 0.001.
+  ///
+  /// [orientation] is the direction of rotation of the cylinder.
+  static Matrix4 cylindricalProjectionTransform({
+    @required double radius,
+    @required double angle,
+    double perspective: 0.001,
+    Axis orientation: Axis.vertical,
+  }) {
+    // Simplified projection matrix.
+    // http://web.iitd.ac.in/~hegde/cad/lecture/L9_persproj.pdf.
+    final Matrix4 projectionMatrix = new Matrix4.identity()
+        ..setEntry(3, 2, -perspective);
+
+    // Simplified camera view matrix.
+    // Basically re-scales to keep object at original size at angle = 0 at
+    // any radius.
+    final Matrix4 viewMatrix = new Matrix4.identity()
+        ..setEntry(2, 3, -radius);
+
+    // First translate the object from the origin of the world by radius in the
+    // z axis and then rotate against the world.
+    final Matrix4 modelMatrix = (
+        orientation == Axis.horizontal
+            ? new Matrix4.rotationY(angle)
+            : new Matrix4.rotationX(angle)
+    ) * new Matrix4.translationValues(0.0, 0.0, radius);
+
+    return projectionMatrix * viewMatrix * modelMatrix;
```

Comment:
Not sure if this is a concern at this point, but according to my count this method creates 7 matrices. However, I'm not sure what to do about without making the API more involved.

Reply:
Got rid of the intermediate ones. Down to 4